driver-core/src/main/com/mongodb/internal/connection/ProtocolHelper.java

```diff
@@ -264,7 +265,16 @@ public static MongoException createSpecialException(final BsonDocument response,
         } else if (isNotPrimaryError(errorCode, errorMessage)) {
             return new MongoNotPrimaryException(response, serverAddress);
         } else if (response.containsKey("writeConcernError")) {
-            return createSpecialException(response.getDocument("writeConcernError"), serverAddress, "errmsg");
+            BsonDocument writeConcernError = response.getDocument("writeConcernError");
```

Comment:
I think this is more complex than it needs to be:

```
            MongoException error = createSpecialException(response.getDocument("writeConcernError"), serverAddress, "errmsg");
            for (BsonValue errorLabel : response.getArray("errorLabels", new BsonArray())) {
                error.addLabel(errorLabel.asString().getValue());
            }
            return error;
```

Reply:
That's so much better.  Thanks.  Added a couple of checks to a conditional protecting the loop.  One is necessary for correctness, the other just to document intent.

Done.